plugin/web/extensions/sidebar/modals/datacatalog/components/content/DatasetsPage/index.tsx

```diff
@@ -62,10 +58,11 @@ const DatasetsPage: React.FC<Props> = ({
     [addedDatasetDataIDs],
   );
 
-  const selectedDataset = useMemo(
-    () => catalog?.find(item => item.dataID !== undefined && item.dataID === selectedDatasetID),
-    [catalog, selectedDatasetID],
-  );
+  const selectedDataset = useMemo(() => {
+    return selectedItem && "dataID" in selectedItem
+      ? catalog?.find(item => item.dataID === selectedItem.dataID)
+      : selectedItem;
+  }, [catalog, selectedItem]);
```

Comment:
Sorry I know you explained this change, but can you explain why its needed still?
Also, I explained the need for checking if it is undefined.

Reply:
@KaWaite , `selectedDataset` doesn't have all the properties needed for the details, so it will fetch the right data from the catalog (it always has been like this).
The edit I made is if the `selectedDataset` is not a `DataCatalogItem` get the `DataCatalogGroup ` itself as now it has a `desc` property.

as for undefined, my mistake I updated it now.

Reply:
Now that I think about, since you consolidated `selectedDatasetID` and `selectedItem`, why do you think `selectedDataset` is even needed? Now we basically have CatalogItem1 already assigned to selectedItem, and then we are searching the catalog again for CatalogItem1 and assigning it to selectedDataset, and then using selectedDataset.
Can't we just use selectedItem as is?
If selectedItem isn't selected at the level where the folder and desc is already added I feel like our logic is wrong somewhere.

Reply:
@KaWaite , it is needed, It is responsible to get the latest state of `DataCatalogItem ` each time, if we remove that part (useMemo) it will break the publishing feature here, it won't change it's status after each publish/unpublish.

Reply:
Then we should use the old selectedDatasetID rather than the selectedItem, right? That way we don't needlessly save the whole dataset object